core/templates/domain/statistics/learner-answer-details-backend-api.service.ts

```diff
@@ -16,52 +16,61 @@
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { HttpClient, HttpResponse } from '@angular/common/http';
+import { Injectable } from '@angular/core';
 
-require('domain/statistics/statistics-domain.constants.ajs.ts');
+import { StatisticsDomainConstants } from
+  'domain/statistics/statistics-domain.constants.ts';
+import { UrlInterpolationService } from
+  'domain/utilities/url-interpolation.service.ts';
 
-angular.module('oppia').factory('LearnerAnswerDetailsBackendApiService', [
-  '$http', '$q', 'UrlInterpolationService', 'SUBMIT_LEARNER_ANSWER_DETAILS_URL',
-  function(
-      $http, $q, UrlInterpolationService, SUBMIT_LEARNER_ANSWER_DETAILS_URL) {
-    var _recordLearnerAnswerDetails = function(
-        expId, stateName, interactionId, answer, answerDetails,
-        successCallback, errorCallback) {
-      var recordLearnerAnswerDetailsUrl = (
-        UrlInterpolationService.interpolateUrl(
-          SUBMIT_LEARNER_ANSWER_DETAILS_URL, {
-            entity_type: 'exploration',
-            entity_id: expId
-          }));
+@Injectable({
+  providedIn: 'root'
+})
+export class LearnerAnswerDetailsBackendApiService {
+  constructor(
+    private httpClient: HttpClient,
+    private urlInterpolationService: UrlInterpolationService) {}
 
-      var payload = {
-        state_name: stateName,
-        interaction_id: interactionId,
-        answer: answer,
-        answer_details: answerDetails
-      };
+  recordLearnerAnswerDetails(
+      explorationId: string, stateName: string, interactionId: string,
+      answer: string, answerDetails: string): Promise<object> {
+    return new Promise((resolve, reject) => {
+      this._recordLearnerAnswerDetails(
+        explorationId, stateName, interactionId, answer, answerDetails,
+        resolve, reject);
+    });
+  }
+
+  private _recordLearnerAnswerDetails(
+      explorationId:string, stateName: string, interactionId: string,
+      answer: string, answerDetails: string, successCallback, errorCallback) {
+    let recordLearnerAnswerDetailsUrl = (
+      this.urlInterpolationService.interpolateUrl(
+        StatisticsDomainConstants.SUBMIT_LEARNER_ANSWER_DETAILS_URL, {
+          entity_type: 'exploration',
+          entity_id: explorationId
+        }));
 
-      $http.put(recordLearnerAnswerDetailsUrl, payload).then(function(
-          response) {
-        if (successCallback) {
+    let payload = {
+      state_name: stateName,
+      interaction_id: interactionId,
+      answer: answer,
+      answer_details: answerDetails
+    };
+
+    this.httpClient.put(recordLearnerAnswerDetailsUrl, payload).toPromise()
+      .then((response: HttpResponse<any>) => {
+        if (response) {
           successCallback();
         }
-      }, function(errorResponse) {
-        if (errorCallback) {
-          errorCallback(errorResponse.data);
-        }
+      }, (errorResponse) => {
+        errorCallback(errorResponse.data);
```

Comment:
This should be just errorResponse, right? (If you remember http calls in Angular return similar object as to what was present in .data in AngularJS)